src/software/SfM/main_openMVG2NVM.cpp

```diff
@@ -0,0 +1,273 @@
+  C_Progress_display my_progress_bar( sfm_data.GetViews().size() );
+  for( Views::const_iterator iter = sfm_data.GetViews().begin();
+       iter != sfm_data.GetViews().end(); ++iter , ++my_progress_bar )
+  {
+    const View * view = iter->second.get();
+
+    if ( !sfm_data.IsPoseAndIntrinsicDefined( view ) )
+    {
+      continue;
+    }
+
+    std::ostringstream padding;
+    padding << std::setw( 4 ) << std::setfill( '0' ) << view->id_view;
+    sOutViewIteratorDirectory = stlplus::folder_append_separator( sOutViewsDirectory ) + "view_" + padding.str() ;
+    sOutViewIteratorDirectory2 = stlplus::folder_append_separator( sOutViewsDirectory2 ) + "view_" + padding.str() ;
+
+    const std::string srcImage = stlplus::create_filespec( sfm_data.s_root_path, view->s_Img_path );
+    const std::string dstImage =
+      stlplus::create_filespec( stlplus::folder_append_separator( sOutViewIteratorDirectory ), "undistorted", "png" );
+    const std::string dstImage2 =
+      stlplus::create_filespec( stlplus::folder_append_separator( sOutViewIteratorDirectory2 ), "undistorted", "png" );
+
+    // Create output dir if not present
+    if ( !stlplus::folder_exists( sOutViewIteratorDirectory ) )
+    {
+      stlplus::folder_create( sOutViewIteratorDirectory );
+    }
+
+
+
+    Intrinsics::const_iterator iterIntrinsic = sfm_data.GetIntrinsics().find( view->id_intrinsic );
+    const IntrinsicBase * cam = iterIntrinsic->second.get();
+    const Pinhole_Intrinsic * pinhole_cam = static_cast<const Pinhole_Intrinsic *>( cam );
+    const double flen = pinhole_cam->focal() ;
+    const Pose3 pose = sfm_data.GetPoseOrDie( view );
+    const Mat3 rotation = pose.rotation();
+    const Vec3 center = pose.center();
+
+    const double Cx = center[0] ;
+    const double Cy = center[1] ;
+    const double Cz = center[2] ;
+    Eigen::Quaterniond q( rotation ) ;
+    const double Qx = q.x() ;
+    const double Qy = q.y() ;
+    const double Qz = q.z() ;
+    const double Qw = q.w() ;
+    double d0 = 0.0 ;
+    // Remove distortion
+    if ( cam->have_disto() )
+    {
+      // Undistort and save the image
+      ReadImage( srcImage.c_str(), &image );
+      UndistortImage( image, cam, image_ud, BLACK );
+      WriteImage( dstImage.c_str(), image_ud );
+    }
+    else // (no distortion)
+    {
+      // If extensions match, copy the PNG image
+      if ( stlplus::extension_part( srcImage ) == "PNG" ||
+           stlplus::extension_part( srcImage ) == "png" )
+      {
+        stlplus::file_copy( srcImage, dstImage );
+      }
+      else
+      {
+        ReadImage( srcImage.c_str(), &image );
+        WriteImage( dstImage.c_str(), image );
+      }
+    }
+
+
+    file << dstImage2 << " " <<
+         flen << " " <<
+
+         Qw << " " <<
+         Qx << " " <<
+         Qy << " " <<
+         Qz << " " <<
+
+         Cx << " " <<
+         Cy << " " <<
+         Cz << " " <<
+
+         d0 << " " <<
+         0 << std::endl ;
```

Comment:
You mention D0 D1 0 before and we can see only D0 0.
Perhaps there is a problem in the comment or there.